src/datasets/hub.py

```diff
@@ -0,0 +1,73 @@
+from itertools import chain
+from typing import Optional, Union
+
+from huggingface_hub import (
+    CommitInfo,
+    CommitOperationAdd,
+    CommitOperationDelete,
+    DatasetCard,
+    DatasetCardData,
+    HfApi,
+    HfFileSystem,
+)
+
+from datasets import config
+from datasets.info import DatasetInfosDict
+from datasets.load import load_dataset_builder
+from datasets.utils.metadata import MetadataConfigs
+
+
+def delete_from_hub(
+    repo_id: str,
+    config_name: Optional[str] = None,
+    revision: Optional[str] = None,
+    token: Optional[Union[bool, str]] = None,
+) -> CommitInfo:
+    operations = []
+    # data_files
+    fs = HfFileSystem(endpoint=config.HF_ENDPOINT, token=token)
+    builder = load_dataset_builder(repo_id, config_name, revision=revision, token=token)
+    for data_file in chain(*builder.config.data_files.values()):
+        operations.append(CommitOperationDelete(path_in_repo=fs.resolve_path(data_file).path_in_repo))
```

Comment:
(nit) maybe add a sanity check to make sure it's a file in the same repo (in case someone defines a dataset with data_files urls from another repo)

Reply:
Done.